src/aio/gateway.rs

```diff
@@ -200,6 +198,7 @@ impl Gateway {
         self.perform_request(
             messages::ADD_PORT_MAPPING_HEADER,
             &messages::format_add_port_mapping_message(
+                self.control_schema.get("AddPortMapping").unwrap(),
```

Comment:
This unwrap is safe?

Reply:
Ah, good catch. If for whatever reason the router does not report a schema for the requested action, then this call to `HashMap.get` will panic.

I rewrote it to instead propagate a dedicated RequestError variant.